modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/diagram/Diagram.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+
+import DiagramFooter from './DiagramFooter';
+import DiagramHeader from './DiagramHeader';
+import ImagePins from './ImagePins';
+
+const Diagram = ({
+	imageSettings,
+	enablePanZoom,
+	enableResetZoom,
+	image,
+	navigationController,
+	spritemap,
+	zoomController,
+}) => {
+	const [resetZoom, setResetZoom] = useState(false);
+	const [zoomInHandler, setZoomInHandler] = useState(false);
+	const [zoomOutHandler, setZoomOutHandler] = useState(false);
+	const [imageState, setImageState] = useState({
+		k: 1,
```

Comment:
Reading this from top-to-bottom so I don't know if this will become clear later, but meaning of `k` is not obvious, at leat to me.

[Edit]: At least at a glance, seems to be a `scale` parameter, is that right?

Reply:
Yep, k is the scale came out from the event handler. but i'm fixing a part that is using it and i'm not sure if it will remain there. I'll advise you asap